backend/shelf-server/src/main/java/com/htec/shelfserver/controller/UserController.java

```diff
@@ -26,16 +29,23 @@ public ResponseEntity getUserById(@AuthenticationUser AuthUser user, @PathVariab
         return ResponseEntity.ok(userService.getUserById(id));
     }
 
-    @PutMapping
-    public String updateUser() {
-        return "Update user is called.";
+    @PutMapping("/{id}")
+    public ResponseEntity<TextResponseMessage> updateUser(@PathVariable Long id,
+                                                          @RequestBody UserUpdateRequestModel userUpdateRequestModel) {
+
+        UserDTO userDTO = UserMapper.INSTANCE.userUpdateRequestModelToUserDto(userUpdateRequestModel);
+        userDTO.setId(id);
+
+        userService.updateUser(userDTO);
+
+        return ResponseEntity.status(HttpStatus.OK).body(new TextResponseMessage("User updated", HttpStatus.OK.value()));
```

Comment:
instead of returning message User updated, consider returning the updated user object?

Reply:
Changed return type